### DISCUSSION TITLE
Checkpoint Meeting

### DISCUSSION THREAD

**Discussion by sanket1211996** [airavata-courses/SuperNova#14] [General]
Today Meeting @ 6 pm -  https://iu.zoom.us/j/9364105744

**Comment by richajha012** [comment]
**For user microservice -** 

Data from UI - email address, session id, timestamp,  query.

**Radar Data Microservice -** 
Technology: Spring Boot...
Functionality : TBD

**Logging service microservices -**
 loggers for all microservices.

**Reply by richajha012** [reply]
https://www.ncdc.noaa.gov/nexradinv/displaygraphs.jsp?mm=01&dd=01&yyyy=2021&product=CCL3N2V&filter=&id=KIND

**Reply by sanket1211996** [reply]
Will take a look

**Comment by sumedhsalvi** [comment]
Todo: Implement a basic prototype of gateway API communication with backend microservices.

**Comment by sanket1211996** [comment]
Checkpoint Meeting - Date 01/24/2022 
Project Plan:

- Stage 1: Basic Repo
- Stage 2: User Interaction Implementation
- Stage 3: Weather Data  Implementation
- Stage 4: Composite System Check: UI + Gateway + Microservices

This seems to be out current implementation plan. Do comment you views.
I will create project planning page & discussion based on this.

**Reply by sumedhsalvi** [reply]
The project plan seems fine.

**Comment by sanket1211996** [comment]
Checkpoint Meeting - Date 01/25/2022

1. Discussed user api data base design changes, needed to update db table design
2. Discussion regarding NEXRAD data and meta-data details
3. Discuss current work states.

Comment down below individual work status:-

**Reply by sanket1211996** [reply]
1. Today I have worked on user dashboard input component, session data table and view port area of the UI project.
2. Also, I have done the analysis on NEXRAD data and its meta-data information.

**Comment by sanket1211996** [comment]
Today Checkpoint Meeting:

Current Folder Structure:
Supernova / Project1-WeatherRadar/ user-api  -     dev-ms-user-api

New Folder Structure:
Supernova / Project1-WeatherRadar/ user-ui  -     dev-ui-user-api
Supernova / Project1-WeatherRadar/ gateway-api -     dev-gateway-api


Weather Radar Processor - AS3 - Plotting

Weather Radar Cache - sessionCache   sessionInfoID, isDataPresent, path - 1,yes, test/

gRPC-

User Api -
userInfo - userID, emailAddress
sessionInfo - id, userid, radstation, date - 1, 10, RDPX,12/01/12

get -  input - emailAddress    output- sessionInfo
post -  input - emailAddress , radStation, date 
            IF.        userInfo - NO - > userinfo-entry  -> sessionInfo-entry , RESPONSE - SUCCESS
            ELSE.  userInfo - YES userinfo-fetch userID - > sessionInfo-entry,  RESPONSE - SUCCESS

POST JSON OBJECT:
{radStation: 'NY', date: '2022-02-04', email: 'sdarwant@iu.edu'}
